tests/unit/browser/test_mhtml.py

```diff
@@ -0,0 +1,277 @@
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20b
+        -----=_qute-UUID
+        Content-Location: http://g.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20g
+        -----=_qute-UUID
+        Content-Location: http://h.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20h
+        -----=_qute-UUID
+        Content-Location: http://i.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20i
+        -----=_qute-UUID
+        Content-Location: http://t.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20t
+        -----=_qute-UUID
+        Content-Location: http://z.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20z
+        -----=_qute-UUID--
+        """)
+
+
+def test_empty_content_type(checker):
+    writer = mhtml.MHTMLWriter(root_content=b'',
+                               content_location='http://example.com/',
+                               content_type='text/plain')
+    writer.add_file('http://example.com/file', b'file content')
+    writer.write_to(checker.fp)
+    checker.expect("""
+        Content-Type: multipart/related; boundary="---=_qute-UUID"
+        MIME-Version: 1.0
+
+        -----=_qute-UUID
+        Content-Location: http://example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+
+        -----=_qute-UUID
+        MIME-Version: 1.0
+        Content-Location: http://example.com/file
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20content
+        -----=_qute-UUID--
+        """)
+
+
+@pytest.mark.parametrize('has_cssutils', [
+    pytest.mark.skipif(mhtml.cssutils is None,
+                       reason="requires cssutils")(True),
+    False,
+], ids=['with_cssutils', 'no_cssutils'])
+@pytest.mark.parametrize('inline, style, expected_urls', [
+    (False, "@import 'default.css'", ['default.css']),
+    (False, '@import "default.css"', ['default.css']),
+    (False, "@import \t 'tabbed.css'", ['tabbed.css']),
+    (False, "@import url('default.css')", ['default.css']),
+    (False, """body {
+    background: url("/bg-img.png")
+    }""", ['/bg-img.png']),
+    (True, 'background: url(folder/file.png) no-repeat', ['folder/file.png']),
+    (True, 'content: url()', []),
+])
+def test_css_url_scanner(monkeypatch, has_cssutils, inline, style,
+                         expected_urls):
+    if not has_cssutils:
+        monkeypatch.setattr('qutebrowser.browser.mhtml.cssutils', None)
```

Comment:
It gets skipped if `cssutils` is not installed:

     pytest.mark.skipif(mhtml.cssutils is None,
                        reason="requires cssutils")(True),

I had the `assert cssutils is not None` before, then went for the skipping part - but I can revert that if you want.